stdlib/jdbc/src/main/java/org/ballerinax/jdbc/statement/AbstractSQLStatement.java

```diff
@@ -196,7 +197,12 @@ Object extractValueFromResultSet(ResultSetMetaData rsMeta, ResultSet rs, int ind
             break;
         case Types.DECIMAL:
         case Types.NUMERIC:
-            value = new DecimalValue(rs.getBigDecimal(index));
+            BigDecimal bigDecimal = rs.getBigDecimal(index);
+            if (bigDecimal == null) {
+                value = 0;
+            } else {
+                value = new DecimalValue(bigDecimal);
+            }
```

Comment:
I suppose this method is called when returning generated keys?
What is the scenario this value became null?
Even if this is null I suppose we should rather return nil instead of returning 0, shouldn't we?

Reply:
Here I got null pointer exception since bigDecimal value is null.
```
Caused by: java.lang.NullPointerException
	at org.ballerinalang.jvm.values.DecimalValue.booleanValue(DecimalValue.java:186)
	at org.ballerinalang.jvm.values.DecimalValue.<init>(DecimalValue.java:53)
	at org.ballerinax.jdbc.statement.AbstractSQLStatement.extractValueFromResultSet(AbstractSQLStatement.java:204)
	at org.ballerinax.jdbc.statement.UpdateStatement.getGeneratedKeys(UpdateStatement.java:112)
	at org.ballerinax.jdbc.statement.UpdateStatement.execute(UpdateStatement.java:86)
	at org.ballerinax.jdbc.actions.Update.nativeUpdate(Update.java:43)
```
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/17116/commits/6ed1145c28fa25e3b6032d098c55f06869ac9d99